spec/requests/documents_spec.rb

```diff
@@ -0,0 +1,165 @@
+require 'rails_helper'
+
+RSpec.shared_examples 'download evidence document' do
+  let(:document) { create :document, external_user: document_owner }
+  let(:document_owner) { external_user }
+  let(:test_url) { 'https://example.com/document.doc#123abc' }
+
+  before do
+    allow(Document).to receive(:find).with(document.to_param).and_return(document)
+    allow(document.document.blob).to receive(:service_url).and_return(test_url)
+    allow(document.converted_preview_document.blob).to receive(:service_url) { |args|
+      "#{test_url}?response-content-disposition=#{args[:disposition] || 'inline'}"
+    }
+  end
+
+  context 'when the document is owned by the logged in user' do
+    it { is_expected.to redirect_to "#{test_url}?response-content-disposition=#{expected_disposition}" }
+  end
+
+  context 'when the document is owned by another user' do
+    let(:document_owner) { create :external_user }
+
+    it { is_expected.to redirect_to external_users_root_url }
+  end
+
+  context 'when the user is not signed in' do
+    before { sign_out user }
+
+    it { is_expected.to redirect_to new_user_session_url }
+  end
+end
+
+RSpec.shared_examples 'failed document upload' do
+  it 'does not create a document' do
+    expect { create_document }.not_to change(Document, :count)
+  end
+
+  it 'returns status unprocessable entity' do
+    create_document
+    expect(response.status).to eq(422)
+  end
+
+  it 'returns errors in response' do
+    create_document
+    expect(JSON.parse(response.body)).to have_key('error')
+  end
+end
+
+RSpec.describe '/documents', type: :request do
```

Comment:
```suggestion
RSpec.describe 'documents', type: :request do
```
OR
```suggestion
RSpec.describe 'Document management', type: :request do
```

I always check what it looks like when you use `rspec -fd` to see the verbose output